src/builder.rs

```diff
@@ -111,6 +111,73 @@ impl<W: Write> Builder<W> {
         append(self.get_mut(), header, &mut data)
     }
 
+    /// Appends the contents of an existing archive to this archive.
+    ///
+    /// This function will iterate through the entries of the supplied archive,
+    /// appending them one by one to the target.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error for any intermittent I/O error which
+    /// occurs when either reading or writing.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::fs::File;
+    /// use tar::{Archive, Builder};
+    ///
+    /// let mut input = Archive::new(File::open("/path/to/existing.tar").unwrap());
+    ///
+    /// let mut output_file = File::create("/path/to/target.tar").unwrap();
+    /// let mut output = Builder::new(output_file);
+    ///
+    /// output.append_archive(&mut input).unwrap();
+    /// output.finish().unwrap();
+    /// ```
+    pub fn append_archive<R: Read>(&mut self, archive: &mut Archive<R>) -> io::Result<()> {
+        let entries = archive.entries()?;
```

Comment:
I think this may want to use `entries_raw` perhaps? That way it'll handle entries-describing-other-entries I think